### PR TITLE
spirv-opt: fix copy-propagate-arrays index opti on structs.

### PR DESCRIPTION
**Pull request by Keenuts**
Copy-propagate-arrays optimization pass would create unused constants,
even if the optimization not completed.
This was caused by the way we handled OpAccessChain squashing: we
only referenced constants, and had to create them upfront.

For now, I couldn't find a path to reproduce this. (Might have one involving RuntimeArrays, but couldn't generate one).
This however will be more visible once #4887 is fixed since we would need to abort while computing the final OpAccessChain.

Signed-off-by: Nathan Gauër <brioche@google.com>

### CODE REVIEW

**Review by s-perron — APPROVED** [review]

File: test/opt/copy_prop_array_test.cpp
Comment (s-perron) [inline test/opt/copy_prop_array_test.cpp:1888]: Generally looks good.  However, this test is a bit long.  Could you try to reduce it?  For example, you should only need a single function.  You could run the inliner to do that for you.  The OpName instructions are not needed. There is a tool (`spirv-reduce`) that could help.

Comment (s-perron) [inline test/opt/copy_prop_array_test.cpp:1888]: Add a comment that explains what this test is testing, possibly referencing the instructions in the spir-v.  It will help if somebody has to update the test later.

**Review by s-perron — CHANGES_REQUESTED** [review]

File: test/opt/copy_prop_array_test.cpp
Comment (Keenuts) [reply test/opt/copy_prop_array_test.cpp:1900]: Done, thanks for pointing this out!

Comment (Keenuts) [reply test/opt/copy_prop_array_test.cpp:1900]: Added

File: test/opt/copy_prop_array_test.cpp
Comment (s-perron) [inline test/opt/copy_prop_array_test.cpp:1895]: The size of the array should be <n>, not 2.

**Review by s-perron — APPROVED** [review]
Looks goot.  Just a minor nit.  The array size in the comment.

File: test/opt/copy_prop_array_test.cpp
Comment (Keenuts) [reply test/opt/copy_prop_array_test.cpp]: ooops, thanks

**Comment by s-perron** [comment]
Looks good, but do not merge until Kokoro runs the presubmits.  It seems to be slow today.

**Comment by Keenuts** [comment]
Thanks for the review!
Kokoro has finished :) (But I do not have push permissions, so cannot press submit)

**Comment by s-perron** [comment]
I don't think the tests ran.  When they run, there are 20 checks.  This only has 2.

**Pull request merged**
